packages/front-end/components/Experiment/ResultsDownloadButton.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+  ExperimentReportResultDimension,
+  ExperimentReportVariation,
+} from "back-end/types/report";
+import React, { useMemo } from "react";
+import { FaFileExport } from "react-icons/fa";
+import { useDefinitions } from "../../services/DefinitionsContext";
+import { ExperimentTableRow, getRisk } from "../../services/experiments";
+import { Parser } from "json2csv";
+
+type CsvRow = {
+  date?: string;
+  dimension?: string;
+  metric?: string;
+  variant?: string;
+  riskOfChoosing?: number;
+  users?: number;
+  count?: number;
+  value?: number;
+  chanceToBeatControl?: number | null;
+  percentChange?: number | null;
+};
+
+export default function ResultsDownloadButton({
+  results,
+  metrics,
+  variations,
+  trackingKey,
+  dimension,
+}: {
+  results?: ExperimentReportResultDimension[];
+  metrics: string[];
+  variations: ExperimentReportVariation[];
+  trackingKey: string;
+  dimension: string;
+}) {
+  const { getMetricById, getDimensionById, ready } = useDefinitions();
+
+  const dimensionName =
+    getDimensionById(dimension)?.name ||
+    dimension?.split(":")?.[1] ||
+    dimension ||
+    null;
+
+  const getRows = () => {
+    const csvRows: CsvRow[] = [];
+
+    if (!results || !variations || !ready) return [];
+
+    results.forEach((result) => {
+      metrics.forEach((m) => {
+        result.variations.forEach((variation, index) => {
+          const metric = getMetricById(m);
+          const row: ExperimentTableRow = {
+            label: metric?.name,
+            metric: metric,
+            rowClass: metric?.inverse ? "inverse" : "",
+            variations: result.variations.map((v) => {
+              return v.metrics[m];
+            }),
+          };
+          const { relativeRisk } = getRisk(index, row);
+          csvRows.push({
+            ...(result.name !== "All" && { [dimensionName]: result.name }),
+            metric: metric?.name,
+            variant: variations[index].name,
+            riskOfChoosing: relativeRisk,
+            users: variation.metrics[m].users,
+            count: variation.metrics[m].value,
+            value: variation.metrics[m].cr,
```

Comment:
I don't know if `count` and `value` as labels makes sense for all metric types.

`stats.value` is the sum of all conversions.  So for binomial metrics, it would be the number of conversions.  For revenue, it would be the total dollar amount of all conversions.

`stats.cr` is the per-user value.  So it's equal to `stats.value / stats.users`.  If a variation had $10,000 total revenue and 1000 users, the "cr" would be 10.

I'm thinking we could name the columns `totalValue` and `perUserValue` or something like that.

Reply:
👍